### PR TITLE
addition of comments and modification of methods

### PR DESCRIPTION
**Pull request by AshBll**

### CODE REVIEW

**Review by maxime-auneau — APPROVED** [review]
ok

**Review by maxime-auneau — DISMISSED** [review]
ok

**Review by FranciscoKevin — DISMISSED** [review]
C'est bon pour ma part

File: src/Controller/GameController.php [hunk 1]
```diff
@@ -10,51 +10,86 @@
...
 
 
+        /**
+         *  If he does not find the troops return an error message and redirect to the game page.
+         */
         foreach ($troops as $troop) {
             if (false === $this->troopManager->insert($troop)) {
                 header("HTTP/1.0 404 Not Found");
                 echo '404 - Page not found';
+                header('Location: /game/play');
```
Comment (RomainClair) [inline src/Controller/GameController.php]: You must not redirect and have an HTTP status code. Choose one

File: src/Controller/GameController.php [hunk 2]
```diff
@@ -10,51 +10,86 @@
...
+    }
+
+    /**
+     *  Check with the try method and capture if there is an error rendered can be displayed otherwise
+     */
+    public function play()
     {
-        $troops = $this->troopManager->selectAll();
-        return $this->twig->render("Game/troop.html.twig", ["troops" => $troops]);
+        $this->troops = $this->troopManager->selectAll();
```
Comment (RomainClair) [inline src/Controller/GameController.php]: You troops must not be a property !

File: src/Model/TroopManager.php [hunk 3]
```diff
@@ -48,7 +48,7 @@ public function insert(Troop $troop)
 
     public function deleteAll()
     {
-        // prepared request
+        // prepared request to the database and verified if
         $truncate = $this->pdo->prepare("TRUNCATE " . self::TABLE);
         if (false == $truncate) {
```
Comment (RomainClair) [inline src/Model/TroopManager.php]: You should use strict comparison whenever it's possible

**Review by RomainClair — CHANGES_REQUESTED** [review]

**Review by maxime-auneau — DISMISSED** [review]
ok

**Review by FranciscoKevin — DISMISSED** [review]
Ok

File: src/Model/TroopManager.php [hunk 4]
```diff
@@ -57,4 +57,16 @@ public function deleteAll()
...
+     *  SQL query allowing to select the force in the troop table and to check if it can be executed.
+     */
+    public function selectTroop()
+    {
+        $select = $this->pdo->prepare("SELECT strength FROM " . self::TABLE . " WHERE id=:id");
+        if ($select->execute()) {
+            return $select->fetch(PDO::FETCH_ASSOC);
+        }
+        return "";
+    }
```
Comment (RomainClair) [inline src/Model/TroopManager.php]: You have a parameters that you don't assign a value to.
Don't return empty string without purpose.
And could you reuse the selectOneById() indead ?

**Review by RomainClair — CHANGES_REQUESTED** [review]

**Review by maxime-auneau — DISMISSED** [review]
ok

**Review by maxime-auneau — DISMISSED** [review]
ok

**Review by maxime-auneau — APPROVED** [review]
ok

**Review by FranciscoKevin — APPROVED** [review]
Ok

**Review by thocle86 — APPROVED** [review]

**Pull request closed without merging**